packages/block-library/src/navigation-link/link-ui.js

```diff
@@ -146,6 +162,12 @@ export function LinkUI( props ) {
 		};
 	}
 
+	const link = {
+		url: props.link.url,
+		opensInNewTab: props.link.opensInNewTab,
+		title: props.link.label && navStripHTML( props.link.label ),
```

Comment:
@MaggieCabrera and I noticed that the implementation in the offcanvas does _not_ strip the HTML prior to utilising in the Link UI. We need to apply the same logic there.

Also we sholud check whether we can then move both over to using the standard `stripHTML` from `@wordpress/dom` or whether there was a particular reason we went with a custom method which is identifcal to `stripHTML` except from that it doesn't run `safeHTML` over the string.

Reply:
I found this comment which suggests moving to the `stripHTML` would be ok

https://github.com/WordPress/gutenberg/pull/33849#discussion_r733860522

Reply:
it also looks like the off-canvas editor is not offering the option to create a new page when it doesn't exist. I don't know if that's intended or not. If not, then that's definitely something to follow up on.

Reply:
> it also looks like the off-canvas editor is not offering the option to create a new page when it doesn't exist. I don't know if that's intended or not. If not, then that's definitely something to follow up on.

It's intentional in the sense that I deliberately removed it. Why? Because including that functionality requires `saveEntityRecord` from `@wordpress/core-data` and we need to avoid coupling that package to block editor. I don't yet have a good way around this...

Reply:
I created https://github.com/WordPress/gutenberg/issues/46193 to track.

Reply:
makes sense, thanks!

Reply:
> I found this comment which suggests moving to the stripHTML would be ok
Should we create an issue for this too?